File: apps/android/app/src/main/java/com/example/shop/feature/product/data/model/ProductDetail.kt
package com.example.shop.feature.product.data.model

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject

@Serializable
data class ProductDetailRequest(val id: String)

@Serializable
data class ProductDetail(
    val id: String,
    val title: String,
    val slug: String,
    val description: String? = null,
    val brand: String? = null,
    val status: String,
    val attributes: JsonObject? = null,
    val minPrice: String? = null,
    val maxPrice: String? = null,
    val totalSales: Int,
    val avgRating: String,
    val reviewCount: Int,
    val createdAt: String,
    val updatedAt: String,
    val images: List<ProductImage>,
    val skus: List<Sku>,
    val categories: List<ProductCategory>,
)

@Serializable
data class ProductImage(
    val id: String,
    val url: String,
    val altText: String? = null,
    val isPrimary: Boolean,
    val sortOrder: Int,
)

@Serializable
data class ProductCategory(
    val id: String,
    val name: String,
    val slug: String,
)

File: apps/ios/Shop/Feature/Product/Data/Model/ProductDetail.swift
import Foundation

struct ProductDetail: Codable, Equatable, Sendable {
    let id: String
    let title: String
    let slug: String
    let description: String?
    let brand: String?
    let status: String
    let attributes: [String: String]?
    let minPrice: String?
    let maxPrice: String?
    let totalSales: Int
    let avgRating: String
    let reviewCount: Int
    let createdAt: String
    let updatedAt: String
    let images: [ProductImage]
    let skus: [Sku]
    let categories: [ProductCategory]
}

struct ProductImage: Codable, Equatable, Identifiable, Sendable {
    let id: String
    let url: String
    let altText: String?
    let isPrimary: Bool
    let sortOrder: Int
}

struct Sku: Codable, Equatable, Identifiable, Sendable {
    let id: String
    let skuCode: String
    let price: String
    let comparePrice: String?
    let stock: Int
    let attributes: [String: String]
    let status: String
}

struct ProductCategory: Codable, Equatable, Sendable {
    let id: String
    let name: String
    let slug: String
}
